### PR TITLE
docs: Remove invalid documentation

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Removed invalid sections in documentation:
- Parts related to tekton.
- Mention of VM proxy annotation.

**Release note**:
```release-note
Removed invalid sections from documentation.
```

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -29,29 +27,11 @@ spec:
...
-
-```
-apiVersion: ssp.kubevirt.io/v1beta1
-kind: SSP
-metadata:
-  name: ssp-sample
-  namespace: kubevirt
-spec: {}
-```
-
```
Comment (jcanocan) [inline docs/configuration.md:54]: Is this intentional? Shouldn't be in a separate PR in any case?

Comment (akrejcir) [reply docs/configuration.md:54]: Yes, this part of documentation is no longer valid.
I will change the commit message and PR's description. IMHO this is a simple change and a separate PR is not needed.

Comment (jcanocan) [reply docs/configuration.md:54]: That will work for me. Thanks!

Comment (akrejcir) [reply docs/configuration.md:54]: done

**Comment by jcanocan** [comment]
Thanks a lot @akrejcir!
/lgtm
